core/src/layers/prometheus.rs

```diff
@@ -93,32 +89,148 @@ use crate::*;
 ///     println!("## Prometheus Metrics");
 ///     println!("{}", String::from_utf8(buffer.clone()).unwrap());
 ///     Ok(())
-/// }
+/// # }
 /// ```
 #[derive(Clone, Debug)]
 pub struct PrometheusLayer {
-    registry: Registry,
+    interceptor: PrometheusInterceptor,
+}
+
+impl PrometheusLayer {
+    /// Create a [`PrometheusLayer`] and register its metrics to the default registry.
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// # use log::debug;
+    /// # use opendal::layers::PrometheusLayer;
+    /// # use opendal::services;
+    /// # use opendal::Operator;
+    /// # use opendal::Result;
+    /// #
+    /// # #[tokio::main]
+    /// # async fn main() -> Result<()> {
+    ///     // Pick a builder and configure it.
+    ///     let builder = services::Memory::default();
+    ///
+    ///     let op = Operator::new(builder)?
+    ///         .layer(PrometheusLayer::register_default().expect("register metrics successfully"))
+    ///         .finish();
+    ///     debug!("operator: {op:?}");
+    ///
+    ///     Ok(())
+    /// # }
+    /// ```
+    pub fn register_default() -> Result<Self> {
```

Comment:
Sorry, I missed this API. I believe it belongs to `PrometheusLayerBuilder`.

---

So, why are we moving all APIs out of `PrometheusLayer`?

For composition. Providing any API at `PrometheusLayer` makes it difficult to integrate with other public APIs that `PrometheusLayerBuilder` offers. For example, if users want to use `register_default` and custom-defined buckets, they will need to repeat the same logic here as well.